it/splunk/src/main/java/com/google/cloud/teleport/it/splunk/DefaultSplunkResourceManager.java

```diff
@@ -0,0 +1,375 @@
+    loginArgs.setHost(this.getHost());
+    loginArgs.setScheme(splunkdScheme);
+    loginArgs.setUsername(username);
+    loginArgs.setPassword(builder.password);
+  }
+
+  /**
+   * Helper method for injecting config information from the builder into the given SplunkContainer.
+   *
+   * @param container The SplunkContainer before config info is injected.
+   * @param builder The DefaultSplunkResourceManager.Builder to extract config info from.
+   * @return The SplunkContainer with all config info injected.
+   */
+  private static SplunkContainer setup(SplunkContainer container, Builder builder) {
+    builder.hecPort = builder.hecPort < 0 ? DEFAULT_SPLUNK_HEC_INTERNAL_PORT : builder.hecPort;
+    builder.splunkdPort =
+        builder.splunkdPort < 0 ? DEFAULT_SPLUNKD_INTERNAL_PORT : builder.splunkdPort;
+    builder.hecToken = builder.hecToken == null ? generateHecToken() : builder.hecToken;
+    builder.password = builder.password == null ? generateSplunkPassword() : builder.password;
+    return container
+        .withDefaultsFile(
+            Transferable.of(
+                String.format(
+                    "splunk:\n"
+                        + "  hec:\n"
+                        + "    enable: True\n"
+                        + "    ssl: %b\n"
+                        + "    port: %d\n"
+                        + "    token: %s",
+                    false, builder.hecPort, builder.hecToken)))
+        .withPassword(builder.password);
+  }
+
+  public static DefaultSplunkResourceManager.Builder builder(String testId) {
+    return new DefaultSplunkResourceManager.Builder(testId);
+  }
+
+  @Override
+  public String getHttpEndpoint() {
+    return String.format("%s://%s:%d", hecScheme, getHost(), hecPort);
+  }
+
+  @Override
+  public String getHecEndpoint() {
+    return getHttpEndpoint() + "/services/collector/event";
+  }
+
+  @Override
+  public String getHecToken() {
+    return hecToken;
+  }
+
+  /**
+   * Helper method for converting the given SplunkEvent into JSON format.
+   *
+   * @param event The SplunkEvent to parse.
+   * @return JSON String.
+   */
+  private static String splunkEventToJson(SplunkEvent event) {
+    return new JSONObject(splunkEventToMap(event)).toString();
+  }
+
+  @Override
+  public synchronized boolean sendHttpEvent(SplunkEvent event) {
+    return sendHttpEvents(List.of(event));
+  }
+
+  @Override
+  public synchronized boolean sendHttpEvents(Collection<SplunkEvent> events) {
+
+    LOG.info("Attempting to send {} events to {}.", events.size(), getHecEndpoint());
+
+    // Construct base HEC request
+    HttpPost httppost = new HttpPost(getHecEndpoint());
+    httppost.addHeader("Authorization", " Splunk " + hecToken);
+
+    // Loop over events and send one-by-one
+    try (CloseableHttpClient httpclient = HttpClientBuilder.create().build()) {
+      events.forEach(event -> {
+        String eventStr = splunkEventToJson(event);
+        LOG.info("Sending HTTP event: " + eventStr);
```

Comment:
nit: using {} is usually more memory efficient / creates less strings in the heap.

```suggestion
        LOG.info("Sending HTTP event: {}", eventStr);
```

Reply:
Done.